### ISSUE TITLE
Problems with Seq2Seq model using RNN layer and GRU cells

### ISSUE THREAD

**Issue by GDrAnimal123** [keras-team/keras#11346]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by GDrAnimal123** [comment]
Basically, This is my implementation:
Encoder:
`class Encoder(object):
    def __init__(self, num_words, embedding_size, state_size, layers=2, dropout_rate=0.1):

        self.encoder_embedding = Embedding(input_dim=num_words,
                                           output_dim=embedding_size,
                                           name='encoder_embedding')

        cells = [GRUCell(state_size, dropout=dropout_rate, recurrent_dropout=dropout_rate) for _ in range(layers)]
        self.rnn_layer = RNN(cells, return_state=True)

    def __call__(self, x):
        # Start the neural network with its input-layer.
        net = x
        # Connect the embedding-layer.
        net = self.encoder_embedding(net)

        # Connect all the GRU-layers.
        net = self.rnn_layer(net)

        # This is the output of the encoder.
        encoder_output, encoder_state = net[0], net[1:]

        return encoder_output, encoder_state`

Decoder:
`class Decoder(object):
    def __init__(self, num_words, embedding_size, state_size, layers=2, dropout_rate=0.1):

        # Different embedding due to different languages
        self.embedding = Embedding(input_dim=num_words,
                                   output_dim=embedding_size,
                                   name='decoder_embedding')

        cells = [GRUCell(state_size, dropout=dropout_rate, recurrent_dropout=dropout_rate) for _ in range(layers)]
        self.rnn_layer = RNN(cells, return_sequences=True)
        # self.batch_norm = BatchNormalization()
        self.output = Dense(num_words,
                            activation='softmax',
                            name='decoder_output')

    def __call__(self, x, initial_state):
        # Start the decoder-network with its input-layer.
        net = x
        # Connect the embedding-layer.
        net = self.embedding(net)
        # Connect all the GRU-layers.
        net = self.rnn_layer(net, initial_state=initial_state)

        # Connect the final dense layer that converts to
        # one-hot encoded arrays.
        net = TimeDistributed(self.output)(net)

        decoder_output = net
        return decoder_output`
